pkg/eval/str/str.go

```diff
@@ -8,6 +8,313 @@ import (
 	"github.com/elves/elvish/pkg/eval"
 )
 
+//elvdoc:fn compare
+//
+// ```elvish
+// str:compare $a $b
+// ```
+//
+// Compare two strings and output an integer that will be 0 if a == b,
+// -1 if a < b, and +1 if a > b.
+//
+// ```elvish-transcript
+// ~> str:compare a a
+// ▶ 0
+// ~> str:compare a b
+// ▶ -1
+// ~> str:compare b a
+// ▶ 1
+// ```
+
+//elvdoc:fn contains
+//
+// ```elvish
+// str:contains $str $substr
+// ```
+//
+// If `$substr` is within `$str` output `$true` else `$false`.
+//
+// ```elvish-transcript
+// ~> str:contains abcd x
+// ▶ $false
+// ~> str:contains abcd bc
+// ▶ $true
+// ```
+
+//elvdoc:fn contains-any
+//
+// ```elvish
+// str:contains-any $str $chars
+// ```
+//
+// If any Unicode code points in `$chars` is within `$str` output `$true` else
+// `$false`.
+//
+// ```elvish-transcript
+// ~> str:contains-any abcd x
+// ▶ $false
+// ~> str:contains-any abcd xby
+// ▶ $true
+// ```
+
+//elvdoc:fn count
+//
+// ```elvish
+// str:count $str $substr
+// ```
+//
+// Output the number of non-overlapping instances of `$substr` in `$s`.
+// If `$substr` is an empty string, output 1 + the number of Unicode code
+// points in `$s`.
+//
+// ```elvish-transcript
+// ~> str:count abcdefabcdef bc
+// ▶ 2
+// ~> str:count abcdef ''
+// ▶ 7
+// ```
+
+//elvdoc:fn equal-fold
+//
+// ```elvish
+// str:equal-fold $str1 $str2
+// ```
+//
+// If `$str1` and `$str2`, interpreted as UTF-8 strings, are equal under
+// Unicode case-folding output `$true` else `$false`.
+//
+// ```elvish-transcript
+// ~> str:equal-fold ABC abc
+// ▶ $true
+// ~> str:equal-fold abc ab
+// ▶ $false
+// ```
+
+//elvdoc:fn has-prefix
+//
+// ```elvish
+// str:has-prefix $str $prefix
+// ```
+//
+// If `$str` begins with `$prefix` output `$true` else `$false`.
+//
+// ```elvish-transcript
+// ~> str:has-prefix abc ab
+// ▶ $true
+// ~> str:has-prefix abc bc
+// ▶ $false
+// ```
+
+//elvdoc:fn has-suffix
+//
+// ```elvish
+// str:has-suffix $str $suffix
+// ```
+//
+// If `$str` ends with `$suffix` output `$true` else `$false`.
+//
+// ```elvish-transcript
+// ~> str:has-suffix abc ab
+// ▶ $false
+// ~> str:has-suffix abc bc
+// ▶ $true
+// ```
+
+//elvdoc:fn index
+//
+// ```elvish
+// str:index $str $substr
+// ```
+//
+// Output the index of the first instance of `$substr` in `$str`, or -1
+// if `$substr` is not present in `$str`.
+
```

Comment:
Missing //. The elvdoc for a function or variable must be a continuous comment block.